wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -850,6 +855,174 @@ instance BuildableSafeGen Wallet where
 instance Buildable [Wallet] where
     build = bprint listJson
 
+
+
+--------------------------------------------------------------------------------
+-- Utxo statistics
+--------------------------------------------------------------------------------
+
+-- | Utxo statistics for the wallet.
+-- | Histogram is composed of bars that represent the bucket. The bucket is tagged by upper bound of a given bucket.
+-- | The bar value corresponds to the number of stakes
+-- | In the future the bar value could be different things:
+-- | (a) sum of stakes in a bucket
+-- | (b) avg or std of stake in a bucket
+-- | (c) topN buckets
+-- | to name a few
+newtype HistogramBar = HistogramBarCount (Text, Word64) deriving (Show, Eq, Ord, Generic)
+
+instance ToSchema HistogramBar where
+  declareNamedSchema = genericDeclareNamedSchema defaultSchemaOptions
+
+instance Arbitrary HistogramBar where
+  arbitrary =
+      let possibleBuckets = fmap show $ (zipWith (\ten toPower -> ten^toPower :: Word64) (repeat (10::Word64)) [(1::Word64)..16]) ++ [45 * (10^(15::Word64))]
+          possibleBars = zipWith (curry HistogramBarCount) possibleBuckets [0..]
+      in elements possibleBars
+
+deriveSafeBuildable ''HistogramBar
+instance BuildableSafeGen HistogramBar where
+    buildSafeGen _ (HistogramBarCount pair) =
+        bprint build pair
+
+
+data UtxoStatistics = UtxoStatistics
+  { theHistogram :: ![HistogramBar]
+  , theAllStakes :: !Word64
+  } deriving (Show, Generic, Ord)
+
+toMap :: [HistogramBar] -> Map Text Word64
+toMap = Map.fromList . map (\(HistogramBarCount (key, val)) -> (key,val))
+
+instance Eq UtxoStatistics where
+    (UtxoStatistics h s) == (UtxoStatistics h' s') = s == s' && toMap h == toMap h'
+
+instance ToJSON UtxoStatistics where
+    toJSON (UtxoStatistics bars allStakes) =
+        let histogramObject = Object . HMS.fromList . map extractBarKey
+            extractBarKey (HistogramBarCount (bound, stake)) = bound .= stake
+        in object [ "histogram" .= histogramObject bars
+                  , "allStakes" .= allStakes ]
+
+instance FromJSON UtxoStatistics where
+    parseJSON (Object v) =
```

Comment:
According to specification of the task there expectation to have:
```  "data": {
    "allStakes": 0,
    "histogram": {
      "10000000000000000": 0,
      "100000000000000": 0,
      "1000000000000": 0,
      "10000000000": 0,
      "100000000": 0,
      "1000000": 0,
      "10000": 0,
      "100": 0,
      "1000000000": 0,
      "100000000000": 0,
      "10": 0,
      "45000000000000000": 0,
      "1000": 0,
      "10000000000000": 0,
      "1000000000000000": 0,
      "100000": 0,
      "10000000": 0
    }
  }
```
rather than
```  "data": {
    "allStakes": 0,
    "histogram": [
        {
             "bucketName" : "10",
             "bucketUpperBound" : 0
        },
 .....
    ]
  }
```